### ISSUE TITLE
[Help]: snaphu.csh error

### ISSUE THREAD

**Issue by swjiaotong-jyliu** [gmtsar/gmtsar#809]
![image](https://github.com/gmtsar/gmtsar/assets/86451834/a0f45a46-254e-4df2-a072-050099585da7)
I choose  gauss_400(left) and gauss_200(right).
I found different jumps in the unwarp.grd.
![image](https://github.com/gmtsar/gmtsar/assets/86451834/4eeeb262-07cc-4b54-8d37-e1d36eccfad6)
The jumps looks like on rivers.
Is any way to improve it?
Thanks.

**Comment by swjiaotong-jyliu** [comment]
Or my grdsample too small？

**Comment by swjiaotong-jyliu** [comment]
![1694493685531](https://github.com/gmtsar/gmtsar/assets/86451834/7bb661fc-6982-49cc-b218-17e783e0ddc1)
I find a low corr line in the pdf.

**Comment by swjiaotong-jyliu** [comment]
I try 
```snaphu.csh 0 0```
it does`t be better.
Then I use ```gmtmath corr.grd 1 ADD 0.5 MUL``` to make corr.grd below 0.5-1
and ```snaphu``` but 
![image](https://github.com/gmtsar/gmtsar/assets/86451834/730044a5-99d1-440c-b0b7-3ada731135a1)

**Comment by Xiaohua-Eric-Xu** [comment]
This is a typical error when one applies landmasking and setting ocean to 0 (by default of snaphu). This is why in your time-series recipe, it is recommended to weakly mask the wet area (leaving some pixels inside), and do a nearest-neighbor interpolation, so radar phase could properly change along the coast. 
<img width="1023" alt="image" src="https://github.com/gmtsar/gmtsar/assets/16844111/cf326ee4-587b-4a11-8e4a-33d1eab5e2f5">

**Comment by mobigroup** [comment]
@Xiaohua-Eric-Xu That’s the problem related to the extremely slow nearest neighbor interpolator tool in GMTSAR - it cannot be used by default as expected.

**Comment by Xiaohua-Eric-Xu** [comment]
> @Xiaohua-Eric-Xu That’s the problem related to the extremely slow nearest neighbor interpolator tool in GMTSAR - it cannot be used by default as expected.

I think the default is snaphu.csh not snaphu_interp.csh. We also tried the interpolator in gdal but somehow it was even slower. Maybe when we get our python scripts build, this could be better resolved.

**Comment by mobigroup** [comment]
Yes, you have the solution (snaphu_interp.csh), but it is not stable due to interpolator problems and cannot be used by default. As I remember, I shared a fast Python interpolator years ago. Regarding your Python scripts, are you still planning to build them on top of your shell scripts? The software's complexity will become unmanageable.

**Comment by Xiaohua-Eric-Xu** [comment]
> Regarding your Python scripts, are you still planning to build them on top of your shell scripts?

I don't think that is the plan. The shell and python scripts will be independent to each other with similar utility habits.

**Comment by mobigroup** [comment]
@Xiaohua-Eric-Xu 
> The shell and python scripts will be independent…

When I reviewed the pull request in the repository, I noticed that the Python wrappers were built on top of shell scripts. This means users have to read and trace both the Python and shell code to identify the related binary tools and their arguments. It looks almost impossible to use and debug even on your side.

**Comment by Xiaohua-Eric-Xu** [comment]
Since we are doing this in a top-down direction, a temporary state will be like that.
